paddle/go/optimizer/optimizer_factory.cc

```diff
@@ -0,0 +1,66 @@
+#include "optimizer_factory.h"
+
+namespace paddle {
+namespace optimizer {
+
+template<class T>
+SGDOptimizer<T>::SGDOptimizer(const ::paddle::OptimizerConfig &config) : ParameterOptimizer<T>(config) {
+  learning_rate = config.learning_rate();
+  momentum = config.momentum();
+  decay = config.decay();
+  nesterov = config.nesterov();
+  lr_decay_a = config.lr_decay_a();
+  lr_decay_b = config.lr_decay_b();
+}
+
+template<class T>
+void SGDOptimizer<T>::destroy() {
```

Comment:
I am not very familiar with C++, just curious why do we need this `destroy` function besides already having destructor? Is this a design pattern, can you give me a link?